{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Data Loading"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import matplotlib.pyplot as plt\n",
    "import matplotlib.cm as cm\n",
    "import numpy as np\n",
    "import seaborn as sns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Loaded variable 'df' from URI: h:\\My Drive\\Pathogenic_Landscape\\data\\absolute\\clinical_research_filtered_combined\\final_filtered_combined.tsv\n",
    "ogdf = pd.read_csv(r'D:\\4bc_Gdrive\\My Drive\\Pathogenic_Landscape\\data\\absolute\\clinical_research_filtered_combined\\final_filtered_combined.tsv', sep='\\t', engine='pyarrow')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Total number of variants = 42,06,676"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(4206676, 15)\n"
     ]
    }
   ],
   "source": [
    "print(ogdf.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.microsoft.datawrangler.viewer.v0+json": {
       "columns": [
        {
         "name": "index",
         "rawType": "int64",
         "type": "integer"
        },
        {
         "name": "Sample_Name",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "CHROM_x",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "POS_x",
         "rawType": "int64",
         "type": "integer"
        },
        {
         "name": "End_x",
         "rawType": "int64",
         "type": "integer"
        },
        {
         "name": "REF_x",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "ALT_x",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "Ref.Gene",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "Func.ensGene",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "ExonicFunc.ensGene",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "AAChange.ensGene",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "Interpro_domain",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "avsnp150",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "CLNDN",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "CLNDISDB",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "clinvar: Clinvar ",
         "rawType": "object",
         "type": "string"
        }
       ],
       "conversionMethod": "pd.DataFrame",
       "ref": "eece6962-b4b2-4e58-8405-c3b7d230dd7d",
       "rows": [
        [
         "0",
         "A549-NCCS-SE8",
         "chr1",
         "1858465",
         "1858466",
         "GGA",
         "G",
         "CFAP74",
         "exonic",
         "frameshift deletion",
         "ENSG00000142609:ENST00000493964:exon12:c.1378_1379del:p.S460fs",
         ".",
         ".",
         ".",
         ".",
         "clinvar: UNK "
        ],
        [
         "1",
         "A549-NCCS-SE8",
         "chr1",
         "1861752",
         "1861761",
         "AGCCCGGCAAG",
         "A",
         "CFAP74",
         "exonic",
         "frameshift deletion",
         "ENSG00000142609:ENST00000493964:exon9:c.875_884del:p.S292fs",
         ".",
         ".",
         ".",
         ".",
         "clinvar: UNK "
        ],
        [
         "2",
         "A549-NCCS-SE8",
         "chr1",
         "2328643",
         "2328643",
         "CG",
         "C",
         "RER1",
         "exonic",
         "frameshift deletion",
         "ENSG00000157916:ENST00000488353:exon2:c.170delG:p.R57fs,ENSG00000157916:ENST00000378512:exon3:c.170delG:p.R57fs,ENSG00000157916:ENST00000378513:exon3:c.170delG:p.R57fs,ENSG00000157916:ENST00000378518:exon3:c.170delG:p.R57fs,ENSG00000157916:ENST00000443438:exon3:c.170delG:p.R57fs,ENSG00000157916:ENST00000605895:exon3:c.170delG:p.R57fs,ENSG00000157916:ENST00000306256:exon4:c.170delG:p.R57fs",
         ".",
         ".",
         ".",
         ".",
         "clinvar: UNK "
        ],
        [
         "3",
         "A549-NCCS-SE8",
         "chr1",
         "2411743",
         "2411743",
         "A",
         "AACAT",
         "PLCH2",
         "exonic",
         "frameshift insertion",
         "ENSG00000149527:ENST00000378486:exon4:c.637_638insACAT:p.M213fs,ENSG00000149527:ENST00000378488:exon4:c.637_638insACAT:p.M213fs,ENSG00000149527:ENST00000419816:exon4:c.637_638insACAT:p.M213fs,ENSG00000149527:ENST00000449969:exon4:c.556_557insACAT:p.M186fs",
         ".",
         ".",
         ".",
         ".",
         "clinvar: UNK "
        ],
        [
         "4",
         "A549-NCCS-SE8",
         "chr1",
         "2433637",
         "2433637",
         "G",
         "GA",
         "PLCH2",
         "exonic",
         "frameshift insertion",
         "ENSG00000149527:ENST00000288766:exon7:c.612_613insA:p.P204fs,ENSG00000149527:ENST00000378488:exon20:c.2640_2641insA:p.P880fs,ENSG00000149527:ENST00000378486:exon21:c.2748_2749insA:p.P916fs,ENSG00000149527:ENST00000419816:exon21:c.2748_2749insA:p.P916fs,ENSG00000149527:ENST00000449969:exon21:c.2667_2668insA:p.P889fs",
         ".",
         ".",
         ".",
         ".",
         "clinvar: UNK "
        ],
        [
         "5",
         "A549-NCCS-SE8",
         "chr1",
         "2525361",
         "2525363",
         "CCGG",
         "C",
         "MMEL1",
         "exonic",
         "nonframeshift deletion",
         "ENSG00000142606:ENST00000502556:exon14:c.1286_1288del:p.429_430del,ENSG00000142606:ENST00000504800:exon18:c.1786_1788del:p.596_596del,ENSG00000142606:ENST00000288709:exon19:c.1730_1732del:p.577_578del,ENSG00000142606:ENST00000378412:exon19:c.1757_1759del:p.586_587del",
         ".",
         ".",
         ".",
         ".",
         "clinvar: UNK "
        ],
        [
         "6",
         "A549-NCCS-SE8",
         "chr1",
         "3409341",
         "3409354",
         "CGGAGGGGGCGGGAG",
         "C",
         "MEGF6",
         "intronic",
         ".",
         ".",
         ".",
         ".",
         ".",
         ".",
         "clinvar: UNK "
        ],
        [
         "7",
         "A549-NCCS-SE8",
         "chr1",
         "3607519",
         "3607519",
         "C",
         "CACA",
         "TP73",
         "intronic",
         ".",
         ".",
         ".",
         ".",
         ".",
         ".",
         "clinvar: UNK "
        ],
        [
         "8",
         "A549-NCCS-SE8",
         "chr1",
         "6593488",
         "6593492",
         "GAGTGA",
         "G",
         "NOL9",
         "exonic",
         "frameshift deletion",
         "ENSG00000162408:ENST00000377705:exon7:c.1085_1089del:p.F362fs",
         ".",
         ".",
         ".",
         ".",
         "clinvar: UNK "
        ],
        [
         "9",
         "A549-NCCS-SE8",
         "chr1",
         "6659171",
         "6659171",
         "G",
         "T",
         "KLHL21",
         "exonic",
         "nonsynonymous SNV",
         "ENSG00000162413:ENST00000377658:exon2:c.C1363A:p.Q455K,ENSG00000162413:ENST00000377663:exon2:c.C1363A:p.Q455K,ENSG00000162413:ENST00000463043:exon2:c.C262A:p.Q88K,ENSG00000162413:ENST00000467612:exon2:c.C262A:p.Q88K,ENSG00000162413:ENST00000496707:exon2:c.C262A:p.Q88K",
         "Kelch-type beta propeller",
         ".",
         ".",
         ".",
         "clinvar: UNK "
        ],
        [
         "10",
         "A549-NCCS-SE8",
         "chr1",
         "10042579",
         "10042579",
         "C",
         "A",
         "NMNAT1",
         "exonic",
         "nonsynonymous SNV",
         "ENSG00000173614:ENST00000377205:exon5:c.C660A:p.D220E,ENSG00000173614:ENST00000462686:exon5:c.C660A:p.D220E",
         "Cytidyltransferase-like domain|Rossmann-like alpha/beta/alpha sandwich fold",
         ".",
         ".",
         ".",
         "clinvar: UNK "
        ],
        [
         "11",
         "A549-NCCS-SE8",
         "chr1",
         "11106665",
         "11106668",
         "AGTCG",
         "A",
         "MASP2",
         "exonic",
         "frameshift deletion",
         "ENSG00000009724:ENST00000400897:exon3:c.357_360del:p.S119fs,ENSG00000009724:ENST00000400898:exon3:c.357_360del:p.S119fs",
         ".",
         ".",
         ".",
         ".",
         "clinvar: UNK "
        ],
        [
         "12",
         "A549-NCCS-SE8",
         "chr1",
         "11206807",
         "11206807",
         "CG",
         "C",
         "MTOR",
         "exonic",
         "frameshift deletion",
         "ENSG00000198793:ENST00000361445:exon32:c.4612delC:p.R1538fs",
         ".",
         ".",
         ".",
         ".",
         "clinvar: UNK "
        ],
        [
         "13",
         "A549-NCCS-SE8",
         "chr1",
         "11318632",
         "11318632",
         "T",
         "TA",
         "MTOR",
         "exonic",
         "frameshift insertion",
         "ENSG00000198793:ENST00000361445:exon3:c.180dupT:p.T61fs",
         ".",
         ".",
         ".",
         ".",
         "clinvar: UNK "
        ],
        [
         "14",
         "A549-NCCS-SE8",
         "chr1",
         "11346128",
         "11346130",
         "TGGG",
         "T",
         "UBIAD1",
         "exonic",
         "nonframeshift deletion",
         "ENSG00000120942:ENST00000376810:exon2:c.957_959del:p.319_320del",
         ".",
         ".",
         ".",
         ".",
         "clinvar: UNK "
        ],
        [
         "15",
         "A549-NCCS-SE8",
         "chr1",
         "15702224",
         "15702225",
         "AGC",
         "A",
         "FHAD1",
         "exonic",
         "frameshift deletion",
         "ENSG00000142621:ENST00000314668:exon10:c.1342_1343del:p.A448fs,ENSG00000142621:ENST00000314740:exon11:c.1396_1397del:p.A466fs,ENSG00000142621:ENST00000529606:exon11:c.1450_1451del:p.A484fs,ENSG00000142621:ENST00000375998:exon26:c.3637_3638del:p.A1213fs,ENSG00000142621:ENST00000417793:exon26:c.3529_3530del:p.A1177fs,ENSG00000142621:ENST00000358897:exon27:c.3637_3638del:p.A1213fs,ENSG00000142621:ENST00000375999:exon27:c.3637_3638del:p.A1213fs",
         ".",
         ".",
         ".",
         ".",
         "clinvar: UNK "
        ],
        [
         "16",
         "A549-NCCS-SE8",
         "chr1",
         "15844889",
         "15844889",
         "C",
         "CAGA",
         "CASP9",
         "exonic",
         "nonframeshift insertion",
         "ENSG00000132906:ENST00000333868:exon2:c.133_134insTCT:p.R45delinsLW,ENSG00000132906:ENST00000348549:exon2:c.133_134insTCT:p.R45delinsLW,ENSG00000132906:ENST00000440484:exon2:c.133_134insTCT:p.R45delinsLW,ENSG00000132906:ENST00000474305:exon2:c.133_134insTCT:p.R45delinsLW,ENSG00000132906:ENST00000546424:exon2:c.133_134insTCT:p.R45delinsLW",
         ".",
         ".",
         ".",
         ".",
         "clinvar: UNK "
        ],
        [
         "17",
         "A549-NCCS-SE8",
         "chr1",
         "16464585",
         "16464585",
         "C",
         "CA",
         "EPHA2",
         "exonic",
         "stopgain",
         "ENSG00000142627:ENST00000358432:exon5:c.1074_1075insT:p.D359_I360delinsX",
         ".",
         ".",
         ".",
         ".",
         "clinvar: UNK "
        ],
        [
         "18",
         "A549-NCCS-SE8",
         "chr1",
         "16891233",
         "16891233",
         "T",
         "C",
         "NBPF1",
         "intronic",
         ".",
         ".",
         ".",
         "rs4120702",
         ".",
         ".",
         "clinvar: UNK "
        ],
        [
         "19",
         "A549-NCCS-SE8",
         "chr1",
         "16891243",
         "16891243",
         "C",
         "G",
         "NBPF1",
         "intronic",
         ".",
         ".",
         ".",
         "rs2289553",
         ".",
         ".",
         "clinvar: UNK "
        ],
        [
         "20",
         "A549-NCCS-SE8",
         "chr1",
         "16907863",
         "16907863",
         "G",
         "C",
         "NBPF1",
         "intronic",
         ".",
         ".",
         ".",
         "rs375442303",
         ".",
         ".",
         "clinvar: UNK "
        ],
        [
         "21",
         "A549-NCCS-SE8",
         "chr1",
         "16948722",
         "16948722",
         "C",
         "T",
         "CROCCP2",
         "ncRNA_intronic",
         ".",
         ".",
         ".",
         "rs390951",
         ".",
         ".",
         "clinvar: UNK "
        ],
        [
         "22",
         "A549-NCCS-SE8",
         "chr1",
         "16955217",
         "16955217",
         "C",
         "T",
         "CROCCP2",
         "ncRNA_intronic",
         ".",
         ".",
         ".",
         "rs4661868",
         ".",
         ".",
         "clinvar: UNK "
        ],
        [
         "23",
         "A549-NCCS-SE8",
         "chr1",
         "16969381",
         "16969381",
         "C",
         "T",
         "CROCCP2",
         "ncRNA_intronic",
         ".",
         ".",
         ".",
         "rs9727604",
         ".",
         ".",
         "clinvar: UNK "
        ],
        [
         "24",
         "A549-NCCS-SE8",
         "chr1",
         "16972134",
         "16972134",
         "C",
         "T",
         "MST1P2",
         "ncRNA_exonic",
         ".",
         ".",
         ".",
         "rs4661404",
         ".",
         ".",
         "clinvar: UNK "
        ],
        [
         "25",
         "A549-NCCS-SE8",
         "chr1",
         "16972920",
         "16972920",
         "C",
         "T",
         "MST1P2",
         "ncRNA_exonic",
         ".",
         ".",
         ".",
         "rs4661882",
         ".",
         ".",
         "clinvar: UNK "
        ],
        [
         "26",
         "A549-NCCS-SE8",
         "chr1",
         "16973606",
         "16973606",
         "G",
         "T",
         "MST1P2",
         "ncRNA_exonic",
         ".",
         ".",
         ".",
         "rs76859932",
         ".",
         ".",
         "clinvar: UNK "
        ],
        [
         "27",
         "A549-NCCS-SE8",
         "chr1",
         "16974511",
         "16974511",
         "G",
         "A",
         "MST1P2",
         "ncRNA_exonic",
         ".",
         ".",
         ".",
         "rs58679574",
         ".",
         ".",
         "clinvar: UNK "
        ],
        [
         "28",
         "A549-NCCS-SE8",
         "chr1",
         "16974592",
         "16974592",
         "C",
         "T",
         "MST1P2",
         "ncRNA_exonic",
         ".",
         ".",
         ".",
         "rs201212576",
         ".",
         ".",
         "clinvar: UNK "
        ],
        [
         "29",
         "A549-NCCS-SE8",
         "chr1",
         "16974637",
         "16974637",
         "T",
         "C",
         "MST1P2",
         "ncRNA_exonic",
         ".",
         ".",
         ".",
         "rs16854791",
         ".",
         ".",
         "clinvar: UNK "
        ],
        [
         "30",
         "A549-NCCS-SE8",
         "chr1",
         "16974662",
         "16974662",
         "C",
         "A",
         "MST1P2",
         "ncRNA_exonic",
         ".",
         ".",
         ".",
         "rs28460410",
         ".",
         ".",
         "clinvar: UNK "
        ],
        [
         "31",
         "A549-NCCS-SE8",
         "chr1",
         "16974869",
         "16974869",
         "G",
         "C",
         "MST1P2",
         "ncRNA_exonic",
         ".",
         ".",
         ".",
         "rs4570422",
         ".",
         ".",
         "clinvar: UNK "
        ],
        [
         "32",
         "A549-NCCS-SE8",
         "chr1",
         "16975045",
         "16975045",
         "G",
         "A",
         "MST1P2",
         "ncRNA_exonic",
         ".",
         ".",
         ".",
         "rs55763726",
         ".",
         ".",
         "clinvar: UNK "
        ],
        [
         "33",
         "A549-NCCS-SE8",
         "chr1",
         "16975994",
         "16975994",
         "T",
         "C",
         "MST1P2",
         "ncRNA_exonic",
         ".",
         ".",
         ".",
         "rs2261801",
         ".",
         ".",
         "clinvar: UNK "
        ],
        [
         "34",
         "A549-NCCS-SE8",
         "chr1",
         "16976032",
         "16976032",
         "A",
         "C",
         "MST1P2",
         "ncRNA_intronic",
         ".",
         ".",
         ".",
         "rs2483047",
         ".",
         ".",
         "clinvar: UNK "
        ],
        [
         "35",
         "A549-NCCS-SE8",
         "chr1",
         "17313065",
         "17313065",
         "C",
         "CACT",
         "ATP13A2",
         "exonic",
         "nonframeshift insertion",
         "ENSG00000159363:ENST00000326735:exon28:c.3297_3298insAGT:p.G1100delinsSG,ENSG00000159363:ENST00000452699:exon28:c.3282_3283insAGT:p.G1095delinsSG",
         ".",
         ".",
         ".",
         ".",
         "clinvar: UNK "
        ],
        [
         "36",
         "A549-NCCS-SE8",
         "chr1",
         "17338247",
         "17338247",
         "G",
         "GAGT",
         "ATP13A2",
         "UTR5",
         ".",
         ".",
         ".",
         ".",
         ".",
         ".",
         "clinvar: UNK "
        ],
        [
         "37",
         "A549-NCCS-SE8",
         "chr1",
         "17749399",
         "17749408",
         "GCGAGTACAAA",
         "G",
         "RCC2",
         "intronic",
         ".",
         ".",
         ".",
         ".",
         ".",
         ".",
         "clinvar: UNK "
        ],
        [
         "38",
         "A549-NCCS-SE8",
         "chr1",
         "17942656",
         "17942656",
         "T",
         "TACA",
         "ARHGEF10L",
         "exonic",
         "nonframeshift insertion",
         "ENSG00000074964:ENST00000375420:exon5:c.68_69insACA:p.V23delinsVQ,ENSG00000074964:ENST00000375415:exon7:c.677_678insACA:p.V226delinsVQ,ENSG00000074964:ENST00000452522:exon8:c.677_678insACA:p.V226delinsVQ,ENSG00000074964:ENST00000361221:exon9:c.794_795insACA:p.V265delinsVQ,ENSG00000074964:ENST00000434513:exon9:c.794_795insACA:p.V265delinsVQ",
         ".",
         ".",
         ".",
         ".",
         "clinvar: UNK "
        ],
        [
         "39",
         "A549-NCCS-SE8",
         "chr1",
         "19446179",
         "19446179",
         "TG",
         "T",
         "UBR4",
         "exonic",
         "frameshift deletion",
         "ENSG00000127481:ENST00000375217:exon70:c.10306delC:p.Q3436fs,ENSG00000127481:ENST00000375226:exon70:c.10255delC:p.Q3419fs,ENSG00000127481:ENST00000375254:exon70:c.10327delC:p.Q3443fs,ENSG00000127481:ENST00000375267:exon70:c.10327delC:p.Q3443fs",
         ".",
         ".",
         ".",
         ".",
         "clinvar: UNK "
        ],
        [
         "40",
         "A549-NCCS-SE8",
         "chr1",
         "19545993",
         "19545994",
         "AAT",
         "A",
         "EMC1-AS1",
         "ncRNA_intronic",
         ".",
         ".",
         ".",
         ".",
         ".",
         ".",
         "clinvar: UNK "
        ],
        [
         "41",
         "A549-NCCS-SE8",
         "chr1",
         "20224170",
         "20224171",
         "CGG",
         "C",
         "OTUD3",
         "intronic",
         ".",
         ".",
         ".",
         ".",
         ".",
         ".",
         "clinvar: UNK "
        ],
        [
         "42",
         "A549-NCCS-SE8",
         "chr1",
         "20517686",
         "20517705",
         "TTTGTACGCCATTTCCGGCCA",
         "T",
         "UBXN10",
         "exonic",
         "frameshift deletion",
         "ENSG00000162543:ENST00000375099:exon2:c.632_651del:p.F211fs",
         ".",
         ".",
         ".",
         ".",
         "clinvar: UNK "
        ],
        [
         "43",
         "A549-NCCS-SE8",
         "chr1",
         "22050508",
         "22050508",
         "C",
         "CACAG",
         "USP48",
         "exonic",
         "frameshift insertion",
         "ENSG00000090686:ENST00000374732:exon1:c.147_148insCTGT:p.A50fs,ENSG00000090686:ENST00000526044:exon4:c.288_289insCTGT:p.A97fs,ENSG00000090686:ENST00000534705:exon4:c.396_397insCTGT:p.A133fs,ENSG00000090686:ENST00000308271:exon12:c.1530_1531insCTGT:p.A511fs,ENSG00000090686:ENST00000400301:exon12:c.1530_1531insCTGT:p.A511fs,ENSG00000090686:ENST00000529637:exon12:c.1527_1528insCTGT:p.A510fs",
         ".",
         ".",
         ".",
         ".",
         "clinvar: UNK "
        ],
        [
         "44",
         "A549-NCCS-SE8",
         "chr1",
         "22155978",
         "22155978",
         "C",
         "CACT",
         "HSPG2",
         "exonic",
         "nonframeshift insertion",
         "ENSG00000142798:ENST00000374695:exon87:c.11889_11890insAGT:p.G3964delinsSG",
         ".",
         ".",
         ".",
         ".",
         "clinvar: UNK "
        ],
        [
         "45",
         "A549-NCCS-SE8",
         "chr1",
         "22178144",
         "22178144",
         "T",
         "TACA",
         "HSPG2",
         "exonic",
         "nonframeshift insertion",
         "ENSG00000142798:ENST00000374695:exon55:c.7052_7053insTGT:p.Q2351delinsHV",
         ".",
         ".",
         ".",
         ".",
         "clinvar: UNK "
        ],
        [
         "46",
         "A549-NCCS-SE8",
         "chr1",
         "23191674",
         "23191674",
         "C",
         "CA",
         "EPHB2",
         "exonic",
         "frameshift insertion",
         "ENSG00000133216:ENST00000374627:exon5:c.1254_1255insA:p.F418fs,ENSG00000133216:ENST00000374630:exon5:c.1272_1273insA:p.F424fs,ENSG00000133216:ENST00000374632:exon5:c.1272_1273insA:p.F424fs,ENSG00000133216:ENST00000400191:exon5:c.1272_1273insA:p.F424fs,ENSG00000133216:ENST00000544305:exon5:c.1272_1273insA:p.F424fs",
         ".",
         ".",
         ".",
         ".",
         "clinvar: UNK "
        ],
        [
         "47",
         "A549-NCCS-SE8",
         "chr1",
         "26384189",
         "26384190",
         "CCG",
         "C",
         "TRIM63",
         "intronic",
         ".",
         ".",
         ".",
         ".",
         ".",
         ".",
         "clinvar: UNK "
        ],
        [
         "48",
         "A549-NCCS-SE8",
         "chr1",
         "27875089",
         "27875089",
         "C",
         "CAGA",
         "AHDC1",
         "exonic",
         "nonframeshift insertion",
         "ENSG00000126705:ENST00000247087:exon5:c.3537_3538insTCT:p.G1180delinsSG,ENSG00000126705:ENST00000374011:exon6:c.3537_3538insTCT:p.G1180delinsSG",
         ".",
         ".",
         ".",
         ".",
         "clinvar: UNK "
        ],
        [
         "49",
         "A549-NCCS-SE8",
         "chr1",
         "28223602",
         "28223602",
         "T",
         "TAGA",
         "RPA2",
         "exonic",
         "nonframeshift insertion",
         "ENSG00000117748:ENST00000313433:exon5:c.702_703insTCT:p.M235delinsSM,ENSG00000117748:ENST00000373909:exon6:c.462_463insTCT:p.M155delinsSM,ENSG00000117748:ENST00000373912:exon6:c.438_439insTCT:p.M147delinsSM,ENSG00000117748:ENST00000444045:exon6:c.450_451insTCT:p.M151delinsSM",
         ".",
         ".",
         ".",
         ".",
         "clinvar: UNK "
        ]
       ],
       "shape": {
        "columns": 15,
        "rows": 500
       }
      },
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sample_Name</th>\n",
       "      <th>CHROM_x</th>\n",
       "      <th>POS_x</th>\n",
       "      <th>End_x</th>\n",
       "      <th>REF_x</th>\n",
       "      <th>ALT_x</th>\n",
       "      <th>Ref.Gene</th>\n",
       "      <th>Func.ensGene</th>\n",
       "      <th>ExonicFunc.ensGene</th>\n",
       "      <th>AAChange.ensGene</th>\n",
       "      <th>Interpro_domain</th>\n",
       "      <th>avsnp150</th>\n",
       "      <th>CLNDN</th>\n",
       "      <th>CLNDISDB</th>\n",
       "      <th>clinvar: Clinvar</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A549-NCCS-SE8</td>\n",
       "      <td>chr1</td>\n",
       "      <td>1858465</td>\n",
       "      <td>1858466</td>\n",
       "      <td>GGA</td>\n",
       "      <td>G</td>\n",
       "      <td>CFAP74</td>\n",
       "      <td>exonic</td>\n",
       "      <td>frameshift deletion</td>\n",
       "      <td>ENSG00000142609:ENST00000493964:exon12:c.1378_...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>clinvar: UNK</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A549-NCCS-SE8</td>\n",
       "      <td>chr1</td>\n",
       "      <td>1861752</td>\n",
       "      <td>1861761</td>\n",
       "      <td>AGCCCGGCAAG</td>\n",
       "      <td>A</td>\n",
       "      <td>CFAP74</td>\n",
       "      <td>exonic</td>\n",
       "      <td>frameshift deletion</td>\n",
       "      <td>ENSG00000142609:ENST00000493964:exon9:c.875_88...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>clinvar: UNK</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A549-NCCS-SE8</td>\n",
       "      <td>chr1</td>\n",
       "      <td>2328643</td>\n",
       "      <td>2328643</td>\n",
       "      <td>CG</td>\n",
       "      <td>C</td>\n",
       "      <td>RER1</td>\n",
       "      <td>exonic</td>\n",
       "      <td>frameshift deletion</td>\n",
       "      <td>ENSG00000157916:ENST00000488353:exon2:c.170del...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>clinvar: UNK</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A549-NCCS-SE8</td>\n",
       "      <td>chr1</td>\n",
       "      <td>2411743</td>\n",
       "      <td>2411743</td>\n",
       "      <td>A</td>\n",
       "      <td>AACAT</td>\n",
       "      <td>PLCH2</td>\n",
       "      <td>exonic</td>\n",
       "      <td>frameshift insertion</td>\n",
       "      <td>ENSG00000149527:ENST00000378486:exon4:c.637_63...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>clinvar: UNK</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A549-NCCS-SE8</td>\n",
       "      <td>chr1</td>\n",
       "      <td>2433637</td>\n",
       "      <td>2433637</td>\n",
       "      <td>G</td>\n",
       "      <td>GA</td>\n",
       "      <td>PLCH2</td>\n",
       "      <td>exonic</td>\n",
       "      <td>frameshift insertion</td>\n",
       "      <td>ENSG00000149527:ENST00000288766:exon7:c.612_61...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>clinvar: UNK</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>495</th>\n",
       "      <td>A549-NCCS-SE8</td>\n",
       "      <td>chr11</td>\n",
       "      <td>134118857</td>\n",
       "      <td>134118860</td>\n",
       "      <td>ATGTG</td>\n",
       "      <td>A</td>\n",
       "      <td>THYN1</td>\n",
       "      <td>intronic</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>clinvar: UNK</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>496</th>\n",
       "      <td>A549-NCCS-SE8</td>\n",
       "      <td>chr12</td>\n",
       "      <td>308079</td>\n",
       "      <td>308079</td>\n",
       "      <td>T</td>\n",
       "      <td>TACA</td>\n",
       "      <td>SLC6A12</td>\n",
       "      <td>exonic</td>\n",
       "      <td>nonframeshift insertion</td>\n",
       "      <td>ENSG00000111181:ENST00000397296:exon7:c.729_73...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>clinvar: UNK</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>497</th>\n",
       "      <td>A549-NCCS-SE8</td>\n",
       "      <td>chr12</td>\n",
       "      <td>432364</td>\n",
       "      <td>432364</td>\n",
       "      <td>GT</td>\n",
       "      <td>G</td>\n",
       "      <td>KDM5A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>frameshift deletion</td>\n",
       "      <td>ENSG00000073614:ENST00000544760:exon7:c.1016de...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>clinvar: UNK</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>498</th>\n",
       "      <td>A549-NCCS-SE8</td>\n",
       "      <td>chr12</td>\n",
       "      <td>527799</td>\n",
       "      <td>527799</td>\n",
       "      <td>T</td>\n",
       "      <td>TA</td>\n",
       "      <td>CCDC77</td>\n",
       "      <td>exonic</td>\n",
       "      <td>frameshift insertion</td>\n",
       "      <td>ENSG00000120647:ENST00000540180:exon3:c.314_31...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>clinvar: UNK</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499</th>\n",
       "      <td>A549-NCCS-SE8</td>\n",
       "      <td>chr12</td>\n",
       "      <td>666023</td>\n",
       "      <td>666035</td>\n",
       "      <td>CGTCGTGCCTGGTG</td>\n",
       "      <td>C</td>\n",
       "      <td>B4GALNT3</td>\n",
       "      <td>exonic</td>\n",
       "      <td>frameshift deletion</td>\n",
       "      <td>ENSG00000139044:ENST00000266383:exon15:c.2371_...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>clinvar: UNK</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>500 rows Ã— 15 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Sample_Name CHROM_x      POS_x      End_x           REF_x  ALT_x  \\\n",
       "0    A549-NCCS-SE8    chr1    1858465    1858466             GGA      G   \n",
       "1    A549-NCCS-SE8    chr1    1861752    1861761     AGCCCGGCAAG      A   \n",
       "2    A549-NCCS-SE8    chr1    2328643    2328643              CG      C   \n",
       "3    A549-NCCS-SE8    chr1    2411743    2411743               A  AACAT   \n",
       "4    A549-NCCS-SE8    chr1    2433637    2433637               G     GA   \n",
       "..             ...     ...        ...        ...             ...    ...   \n",
       "495  A549-NCCS-SE8   chr11  134118857  134118860           ATGTG      A   \n",
       "496  A549-NCCS-SE8   chr12     308079     308079               T   TACA   \n",
       "497  A549-NCCS-SE8   chr12     432364     432364              GT      G   \n",
       "498  A549-NCCS-SE8   chr12     527799     527799               T     TA   \n",
       "499  A549-NCCS-SE8   chr12     666023     666035  CGTCGTGCCTGGTG      C   \n",
       "\n",
       "     Ref.Gene Func.ensGene       ExonicFunc.ensGene  \\\n",
       "0      CFAP74       exonic      frameshift deletion   \n",
       "1      CFAP74       exonic      frameshift deletion   \n",
       "2        RER1       exonic      frameshift deletion   \n",
       "3       PLCH2       exonic     frameshift insertion   \n",
       "4       PLCH2       exonic     frameshift insertion   \n",
       "..        ...          ...                      ...   \n",
       "495     THYN1     intronic                        .   \n",
       "496   SLC6A12       exonic  nonframeshift insertion   \n",
       "497     KDM5A       exonic      frameshift deletion   \n",
       "498    CCDC77       exonic     frameshift insertion   \n",
       "499  B4GALNT3       exonic      frameshift deletion   \n",
       "\n",
       "                                      AAChange.ensGene Interpro_domain  \\\n",
       "0    ENSG00000142609:ENST00000493964:exon12:c.1378_...               .   \n",
       "1    ENSG00000142609:ENST00000493964:exon9:c.875_88...               .   \n",
       "2    ENSG00000157916:ENST00000488353:exon2:c.170del...               .   \n",
       "3    ENSG00000149527:ENST00000378486:exon4:c.637_63...               .   \n",
       "4    ENSG00000149527:ENST00000288766:exon7:c.612_61...               .   \n",
       "..                                                 ...             ...   \n",
       "495                                                  .               .   \n",
       "496  ENSG00000111181:ENST00000397296:exon7:c.729_73...               .   \n",
       "497  ENSG00000073614:ENST00000544760:exon7:c.1016de...               .   \n",
       "498  ENSG00000120647:ENST00000540180:exon3:c.314_31...               .   \n",
       "499  ENSG00000139044:ENST00000266383:exon15:c.2371_...               .   \n",
       "\n",
       "    avsnp150 CLNDN CLNDISDB clinvar: Clinvar   \n",
       "0          .     .        .     clinvar: UNK   \n",
       "1          .     .        .     clinvar: UNK   \n",
       "2          .     .        .     clinvar: UNK   \n",
       "3          .     .        .     clinvar: UNK   \n",
       "4          .     .        .     clinvar: UNK   \n",
       "..       ...   ...      ...               ...  \n",
       "495        .     .        .     clinvar: UNK   \n",
       "496        .     .        .     clinvar: UNK   \n",
       "497        .     .        .     clinvar: UNK   \n",
       "498        .     .        .     clinvar: UNK   \n",
       "499        .     .        .     clinvar: UNK   \n",
       "\n",
       "[500 rows x 15 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ogdf.head(n=500)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1675\n"
     ]
    }
   ],
   "source": [
    "# Number of unique samples\n",
    "unique_sample_count = ogdf['Sample_Name'].nunique()\n",
    "\n",
    "print(unique_sample_count)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Clinvar Variant Annotations and their count"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "clinvar: Clinvar \n",
       "clinvar: UNK                                                                  4166471\n",
       "clinvar: Uncertain_significance                                                 23205\n",
       "clinvar: Conflicting_interpretations_of_pathogenicity                           10627\n",
       "clinvar: Pathogenic                                                              2420\n",
       "clinvar: not_provided                                                            1281\n",
       "clinvar: Likely_pathogenic                                                       1007\n",
       ".                                                                                 510\n",
       "clinvar: Pathogenic/Likely_pathogenic                                             448\n",
       "clinvar: other                                                                    344\n",
       "clinvar: drug_response                                                            133\n",
       "clinvar: Conflicting_interpretations_of_pathogenicity,_risk_factor                 58\n",
       "clinvar: risk_factor                                                               49\n",
       "clinvar: Likely_pathogenic,_risk_factor                                            21\n",
       "clinvar: Affects                                                                   18\n",
       "clinvar: Pathogenic,_risk_factor                                                   16\n",
       "clinvar: Pathogenic/Likely_pathogenic,_other                                       15\n",
       "clinvar: Pathogenic,_Affects                                                       12\n",
       "clinvar: Conflicting_interpretations_of_pathogenicity,_other                        8\n",
       "clinvar: Pathogenic/Likely_pathogenic,_drug_response                                7\n",
       "clinvar: Conflicting_interpretations_of_pathogenicity,_protective                   6\n",
       "clinvar: association                                                                4\n",
       "clinvar: Pathogenic,_other                                                          4\n",
       "clinvar: Pathogenic/Likely_pathogenic,_risk_factor                                  4\n",
       "clinvar: Conflicting_interpretations_of_pathogenicity,_other,_risk_factor           3\n",
       "clinvar: Conflicting_interpretations_of_pathogenicity,_Affects,_other               2\n",
       "clinvar: Likely_pathogenic,_other                                                   1\n",
       "clinvar: Likely_pathogenic,_drug_response                                           1\n",
       "clinvar: Pathogenic,_protective                                                     1\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ogdf['clinvar: Clinvar '].value_counts()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "---\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "#--- Hotspot Variant Markup ---#\n",
    "# Load the text file with hotspot variants\n",
    "hotspot_source_path = r\"D:\\4bc_Gdrive\\My Drive\\Pathogenic_Landscape\\assets\\hotspot_v14.txt\"\n",
    "hotspot_source_df = pd.read_csv(hotspot_source_path, sep=\"\\t\")\n",
    "\n",
    "# Extract relevant columns from hotspot data\n",
    "hotspot_source_variants = set(zip(hotspot_source_df[\"#CHROM\"], hotspot_source_df[\"POS\"], hotspot_source_df[\"REF\"], hotspot_source_df[\"ALT\"]))\n",
    "\n",
    "# Label rows based on hotspot variants\n",
    "ogdf[\"Hotspot\"] = ogdf.apply(lambda row: 1 if (row['CHROM_x'], row['POS_x'], row['REF_x'], row['ALT_x']) in hotspot_source_variants else 0, axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "#--- Oncogene and TSG Gene Markup ---#\n",
    "# Load the text file with oncogene and tumor suppressor gene info obtained from OncoKB\n",
    "oncog_tsg_source_path = r\"D:\\4bc_Gdrive\\My Drive\\Pathogenic_Landscape\\assets\\Oncogene_TSG.txt\"\n",
    "oncog_tsg_source_df = pd.read_csv(oncog_tsg_source_path, sep=\"\\t\")\n",
    "\n",
    "# Convert to a dictionary for quick lookup\n",
    "oncog_tsg_dict = dict(zip(oncog_tsg_source_df[\"Hugo Symbol\"], oncog_tsg_source_df[\"OncoG_TSG\"]))\n",
    "\n",
    "# Identify the gene column\n",
    "gene_column = \"Ref.Gene\"  # Assuming this column contains gene names\n",
    "\n",
    "if gene_column in ogdf.columns:\n",
    "    ogdf[\"OncoG_TSG\"] = ogdf[gene_column].map(oncog_tsg_dict).fillna(\"\")\n",
    "else:\n",
    "    raise ValueError(f\"The required column '{gene_column}' is missing in the dataset.\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "#--- Load the kinase gene list ---#\n",
    "kinase_gene_file = r\"D:\\4bc_Gdrive\\My Drive\\Pathogenic_Landscape\\assets\\Kinase_Genes.txt\"\n",
    "with open(kinase_gene_file, \"r\") as f:\n",
    "    kinase_genes = set(gene.strip() for gene in f.readlines())\n",
    "\n",
    "# Add the \"Kinase\" column\n",
    "ogdf[\"Kinase\"] = ogdf[\"Ref.Gene\"].apply(lambda gene: 1 if gene in kinase_genes else 0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(4202661, 18)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "1669"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#--- Removing Non-Cancer Samples ---#\n",
    "# List of non-cancer samples to remove\n",
    "samples_to_remove = [\n",
    "    'IN-423-TLRA-F-Merged-IE', '005N_IE', '006N_IE', '009N_IE', '010N_IE', '011N_IE']\n",
    "# Drop rows where 'Sample_Name' column contains any of the values in the list\n",
    "df = ogdf[~ogdf['Sample_Name'].isin(samples_to_remove)]\n",
    "print(df.shape)\n",
    "df['Sample_Name'].nunique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "#--- Cancer info Markup ---#\n",
    "# Load the text file with curated cancer info\n",
    "cancer_source_path = r\"D:\\4bc_Gdrive\\My Drive\\Pathogenic_Landscape\\assets\\Absolute Cancer\\Cohort_Cancer.txt\"\n",
    "cancer_source_df = pd.read_csv(cancer_source_path, sep=\"\\t\")\n",
    "\n",
    "# Merge the cancer information based on the Sample_Name column\n",
    "df = df.merge(cancer_source_df, on=\"Sample_Name\", how=\"left\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "#--- Removing N/A Cancer Samples ---#\n",
    "df = df.dropna(subset=['Cancer'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.microsoft.datawrangler.viewer.v0+json": {
       "columns": [
        {
         "name": "index",
         "rawType": "int64",
         "type": "integer"
        },
        {
         "name": "Cancer",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "Unique_Sample_Count",
         "rawType": "int64",
         "type": "integer"
        }
       ],
       "conversionMethod": "pd.DataFrame",
       "ref": "8abb5386-a2ab-41aa-ae4f-dbd4fb4b3236",
       "rows": [
        [
         "0",
         "Brain",
         "11"
        ],
        [
         "1",
         "Breast",
         "109"
        ],
        [
         "2",
         "Cervix",
         "13"
        ],
        [
         "3",
         "Esophagus",
         "25"
        ],
        [
         "4",
         "Gallbladder",
         "61"
        ],
        [
         "5",
         "Head and Neck",
         "68"
        ],
        [
         "6",
         "Kidney",
         "13"
        ],
        [
         "7",
         "Large Intestine",
         "111"
        ],
        [
         "8",
         "Liver and intrahepatic bile ducts",
         "39"
        ],
        [
         "9",
         "Lung",
         "194"
        ],
        [
         "10",
         "Oral",
         "56"
        ],
        [
         "11",
         "Others",
         "52"
        ],
        [
         "12",
         "Ovary",
         "101"
        ],
        [
         "13",
         "Pancreas",
         "51"
        ],
        [
         "14",
         "Prostate",
         "35"
        ],
        [
         "15",
         "Rectum",
         "35"
        ],
        [
         "16",
         "Sarcoma",
         "77"
        ],
        [
         "17",
         "Stomach",
         "140"
        ],
        [
         "18",
         "Thyroid",
         "121"
        ],
        [
         "19",
         "Unknown primary",
         "23"
        ],
        [
         "20",
         "Urinary Bladder",
         "11"
        ],
        [
         "21",
         "Uterus",
         "29"
        ]
       ],
       "shape": {
        "columns": 2,
        "rows": 22
       }
      },
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Cancer</th>\n",
       "      <th>Unique_Sample_Count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Brain</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Breast</td>\n",
       "      <td>109</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Cervix</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Esophagus</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Gallbladder</td>\n",
       "      <td>61</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Head and Neck</td>\n",
       "      <td>68</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Kidney</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Large Intestine</td>\n",
       "      <td>111</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>Liver and intrahepatic bile ducts</td>\n",
       "      <td>39</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Lung</td>\n",
       "      <td>194</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>Oral</td>\n",
       "      <td>56</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>Others</td>\n",
       "      <td>52</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>Ovary</td>\n",
       "      <td>101</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>Pancreas</td>\n",
       "      <td>51</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>Prostate</td>\n",
       "      <td>35</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>Rectum</td>\n",
       "      <td>35</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>Sarcoma</td>\n",
       "      <td>77</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>Stomach</td>\n",
       "      <td>140</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>Thyroid</td>\n",
       "      <td>121</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>Unknown primary</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>Urinary Bladder</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>Uterus</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                               Cancer  Unique_Sample_Count\n",
       "0                               Brain                   11\n",
       "1                              Breast                  109\n",
       "2                              Cervix                   13\n",
       "3                           Esophagus                   25\n",
       "4                         Gallbladder                   61\n",
       "5                       Head and Neck                   68\n",
       "6                              Kidney                   13\n",
       "7                     Large Intestine                  111\n",
       "8   Liver and intrahepatic bile ducts                   39\n",
       "9                                Lung                  194\n",
       "10                               Oral                   56\n",
       "11                             Others                   52\n",
       "12                              Ovary                  101\n",
       "13                           Pancreas                   51\n",
       "14                           Prostate                   35\n",
       "15                             Rectum                   35\n",
       "16                            Sarcoma                   77\n",
       "17                            Stomach                  140\n",
       "18                            Thyroid                  121\n",
       "19                    Unknown primary                   23\n",
       "20                    Urinary Bladder                   11\n",
       "21                             Uterus                   29"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Cancer Counts\n",
    "df.groupby('Cancer')['Sample_Name'].nunique().reset_index(name='Unique_Sample_Count')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Mapping Dictionary\n",
    "cancer_mapping = {\n",
    "    \"Brain\": \"BrC\",\n",
    "    \"Breast\": \"BC\",\n",
    "    \"Cervix\": \"CC\",\n",
    "    \"Esophagus\": \"EC\",\n",
    "    \"Gallbladder\": \"GBC\",\n",
    "    \"Head and Neck\": \"HNC\",\n",
    "    \"Kidney\": \"KC\",\n",
    "    \"Large Intestine\": \"LIC\",\n",
    "    \"Liver and intrahepatic bile ducts\": \"LBC\",\n",
    "    \"Lung\": \"LC\",\n",
    "    \"Oral\": \"OC\",\n",
    "    \"Others\": \"Others\",\n",
    "    \"Ovary\": \"OvC\",\n",
    "    \"Pancreas\": \"PC\",\n",
    "    \"Prostate\": \"PrC\",\n",
    "    \"Rectum\": \"RC\",\n",
    "    \"Sarcoma\": \"SC\",\n",
    "    \"Stomach\": \"StC\",\n",
    "    \"Thyroid\": \"TC\",\n",
    "    \"Unknown primary\": \"UPC\",\n",
    "    \"Urinary Bladder\": \"UBC\",\n",
    "    \"Uterus\": \"UtC\"\n",
    "}\n",
    "\n",
    "# Apply mapping\n",
    "df[\"Cancer_Short\"] = df[\"Cancer\"].map(cancer_mapping)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.microsoft.datawrangler.viewer.v0+json": {
       "columns": [
        {
         "name": "index",
         "rawType": "int64",
         "type": "integer"
        },
        {
         "name": "Cancer_Short",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "Unique_Sample_Count",
         "rawType": "int64",
         "type": "integer"
        }
       ],
       "conversionMethod": "pd.DataFrame",
       "ref": "213f5b58-a374-44aa-872b-e00ea5874bce",
       "rows": [
        [
         "0",
         "BC",
         "109"
        ],
        [
         "1",
         "BrC",
         "11"
        ],
        [
         "2",
         "CC",
         "13"
        ],
        [
         "3",
         "EC",
         "25"
        ],
        [
         "4",
         "GBC",
         "61"
        ],
        [
         "5",
         "HNC",
         "68"
        ],
        [
         "6",
         "KC",
         "13"
        ],
        [
         "7",
         "LBC",
         "39"
        ],
        [
         "8",
         "LC",
         "194"
        ],
        [
         "9",
         "LIC",
         "111"
        ],
        [
         "10",
         "OC",
         "56"
        ],
        [
         "11",
         "Others",
         "52"
        ],
        [
         "12",
         "OvC",
         "101"
        ],
        [
         "13",
         "PC",
         "51"
        ],
        [
         "14",
         "PrC",
         "35"
        ],
        [
         "15",
         "RC",
         "35"
        ],
        [
         "16",
         "SC",
         "77"
        ],
        [
         "17",
         "StC",
         "140"
        ],
        [
         "18",
         "TC",
         "121"
        ],
        [
         "19",
         "UBC",
         "11"
        ],
        [
         "20",
         "UPC",
         "23"
        ],
        [
         "21",
         "UtC",
         "29"
        ]
       ],
       "shape": {
        "columns": 2,
        "rows": 22
       }
      },
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Cancer_Short</th>\n",
       "      <th>Unique_Sample_Count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BC</td>\n",
       "      <td>109</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BrC</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CC</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>EC</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GBC</td>\n",
       "      <td>61</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>HNC</td>\n",
       "      <td>68</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>KC</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>LBC</td>\n",
       "      <td>39</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>LC</td>\n",
       "      <td>194</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>LIC</td>\n",
       "      <td>111</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>OC</td>\n",
       "      <td>56</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>Others</td>\n",
       "      <td>52</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>OvC</td>\n",
       "      <td>101</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>PC</td>\n",
       "      <td>51</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>PrC</td>\n",
       "      <td>35</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>RC</td>\n",
       "      <td>35</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>SC</td>\n",
       "      <td>77</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>StC</td>\n",
       "      <td>140</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>TC</td>\n",
       "      <td>121</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>UBC</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>UPC</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>UtC</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Cancer_Short  Unique_Sample_Count\n",
       "0            BC                  109\n",
       "1           BrC                   11\n",
       "2            CC                   13\n",
       "3            EC                   25\n",
       "4           GBC                   61\n",
       "5           HNC                   68\n",
       "6            KC                   13\n",
       "7           LBC                   39\n",
       "8            LC                  194\n",
       "9           LIC                  111\n",
       "10           OC                   56\n",
       "11       Others                   52\n",
       "12          OvC                  101\n",
       "13           PC                   51\n",
       "14          PrC                   35\n",
       "15           RC                   35\n",
       "16           SC                   77\n",
       "17          StC                  140\n",
       "18           TC                  121\n",
       "19          UBC                   11\n",
       "20          UPC                   23\n",
       "21          UtC                   29"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Cancer Counts\n",
    "df.groupby('Cancer_Short')['Sample_Name'].nunique().reset_index(name='Unique_Sample_Count')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_cancercounts = df.groupby('Cancer_Short')['Sample_Name'].nunique().reset_index(name='Unique_Sample_Count')\n",
    "df_cancercounts.to_clipboard()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1375\n"
     ]
    }
   ],
   "source": [
    "total_unique_samples_in_df = df['Sample_Name'].nunique()\n",
    "print(total_unique_samples_in_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "print(df.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.microsoft.datawrangler.viewer.v0+json": {
       "columns": [
        {
         "name": "index",
         "rawType": "int64",
         "type": "integer"
        },
        {
         "name": "Sample_Name",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "CHROM_x",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "POS_x",
         "rawType": "int64",
         "type": "integer"
        },
        {
         "name": "End_x",
         "rawType": "int64",
         "type": "integer"
        },
        {
         "name": "REF_x",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "ALT_x",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "Ref.Gene",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "Func.ensGene",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "ExonicFunc.ensGene",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "AAChange.ensGene",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "Interpro_domain",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "avsnp150",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "CLNDN",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "CLNDISDB",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "clinvar: Clinvar ",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "Hotspot",
         "rawType": "int64",
         "type": "integer"
        },
        {
         "name": "OncoG_TSG",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "Kinase",
         "rawType": "int64",
         "type": "integer"
        },
        {
         "name": "Cancer",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "Cancer_Short",
         "rawType": "object",
         "type": "string"
        }
       ],
       "conversionMethod": "pd.DataFrame",
       "ref": "ee7e9a0d-4822-4b14-ade2-9afae21d8739",
       "rows": [
        [
         "12012",
         "IN-423-TKKC-F-Merged",
         "chr1",
         "1374966",
         "1374966",
         "TG",
         "T",
         "VWA1",
         "exonic",
         "frameshift deletion",
         "ENSG00000179403:ENST00000404702:exon3:c.501delG:p.V167fs,ENSG00000179403:ENST00000476993:exon3:c.1137delG:p.V379fs",
         ".",
         ".",
         ".",
         ".",
         "clinvar: UNK ",
         "0",
         "",
         "0",
         "Lung",
         "LC"
        ],
        [
         "12013",
         "IN-423-TKKC-F-Merged",
         "chr1",
         "1581065",
         "1581065",
         "G",
         "C",
         "CDK11B",
         "intronic",
         ".",
         ".",
         ".",
         "rs77085542",
         ".",
         ".",
         "clinvar: UNK ",
         "0",
         "",
         "0",
         "Lung",
         "LC"
        ],
        [
         "12014",
         "IN-423-TKKC-F-Merged",
         "chr1",
         "1961365",
         "1961365",
         "G",
         "T",
         "GABRD",
         "intronic",
         ".",
         ".",
         ".",
         ".",
         ".",
         ".",
         "clinvar: UNK ",
         "0",
         "",
         "0",
         "Lung",
         "LC"
        ],
        [
         "12015",
         "IN-423-TKKC-F-Merged",
         "chr1",
         "3102848",
         "3102848",
         "C",
         "T",
         "PRDM16",
         "exonic",
         "nonsynonymous SNV",
         "ENSG00000142611:ENST00000270722:exon2:c.C197T:p.S66L,ENSG00000142611:ENST00000378391:exon2:c.C197T:p.S66L,ENSG00000142611:ENST00000378398:exon2:c.C197T:p.S66L,ENSG00000142611:ENST00000441472:exon2:c.C197T:p.S66L,ENSG00000142611:ENST00000442529:exon2:c.C197T:p.S66L,ENSG00000142611:ENST00000511072:exon2:c.C197T:p.S66L,ENSG00000142611:ENST00000514189:exon2:c.C197T:p.S66L",
         ".",
         "rs374012976",
         "Left_ventricular_noncompaction_8",
         "MedGen:C3809288,OMIM:615373",
         "clinvar: Uncertain_significance ",
         "0",
         "",
         "0",
         "Lung",
         "LC"
        ],
        [
         "12016",
         "IN-423-TKKC-F-Merged",
         "chr1",
         "6515380",
         "6515380",
         "G",
         "T",
         "ESPN",
         "intronic",
         ".",
         ".",
         ".",
         "rs75827056",
         ".",
         ".",
         "clinvar: UNK ",
         "0",
         "",
         "0",
         "Lung",
         "LC"
        ]
       ],
       "shape": {
        "columns": 20,
        "rows": 5
       }
      },
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sample_Name</th>\n",
       "      <th>CHROM_x</th>\n",
       "      <th>POS_x</th>\n",
       "      <th>End_x</th>\n",
       "      <th>REF_x</th>\n",
       "      <th>ALT_x</th>\n",
       "      <th>Ref.Gene</th>\n",
       "      <th>Func.ensGene</th>\n",
       "      <th>ExonicFunc.ensGene</th>\n",
       "      <th>AAChange.ensGene</th>\n",
       "      <th>Interpro_domain</th>\n",
       "      <th>avsnp150</th>\n",
       "      <th>CLNDN</th>\n",
       "      <th>CLNDISDB</th>\n",
       "      <th>clinvar: Clinvar</th>\n",
       "      <th>Hotspot</th>\n",
       "      <th>OncoG_TSG</th>\n",
       "      <th>Kinase</th>\n",
       "      <th>Cancer</th>\n",
       "      <th>Cancer_Short</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>12012</th>\n",
       "      <td>IN-423-TKKC-F-Merged</td>\n",
       "      <td>chr1</td>\n",
       "      <td>1374966</td>\n",
       "      <td>1374966</td>\n",
       "      <td>TG</td>\n",
       "      <td>T</td>\n",
       "      <td>VWA1</td>\n",
       "      <td>exonic</td>\n",
       "      <td>frameshift deletion</td>\n",
       "      <td>ENSG00000179403:ENST00000404702:exon3:c.501del...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>clinvar: UNK</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td>Lung</td>\n",
       "      <td>LC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12013</th>\n",
       "      <td>IN-423-TKKC-F-Merged</td>\n",
       "      <td>chr1</td>\n",
       "      <td>1581065</td>\n",
       "      <td>1581065</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>CDK11B</td>\n",
       "      <td>intronic</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>rs77085542</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>clinvar: UNK</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td>Lung</td>\n",
       "      <td>LC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12014</th>\n",
       "      <td>IN-423-TKKC-F-Merged</td>\n",
       "      <td>chr1</td>\n",
       "      <td>1961365</td>\n",
       "      <td>1961365</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>GABRD</td>\n",
       "      <td>intronic</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>clinvar: UNK</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td>Lung</td>\n",
       "      <td>LC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12015</th>\n",
       "      <td>IN-423-TKKC-F-Merged</td>\n",
       "      <td>chr1</td>\n",
       "      <td>3102848</td>\n",
       "      <td>3102848</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>PRDM16</td>\n",
       "      <td>exonic</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>ENSG00000142611:ENST00000270722:exon2:c.C197T:...</td>\n",
       "      <td>.</td>\n",
       "      <td>rs374012976</td>\n",
       "      <td>Left_ventricular_noncompaction_8</td>\n",
       "      <td>MedGen:C3809288,OMIM:615373</td>\n",
       "      <td>clinvar: Uncertain_significance</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td>Lung</td>\n",
       "      <td>LC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12016</th>\n",
       "      <td>IN-423-TKKC-F-Merged</td>\n",
       "      <td>chr1</td>\n",
       "      <td>6515380</td>\n",
       "      <td>6515380</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>ESPN</td>\n",
       "      <td>intronic</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>rs75827056</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>clinvar: UNK</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td>Lung</td>\n",
       "      <td>LC</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                Sample_Name CHROM_x    POS_x    End_x REF_x ALT_x Ref.Gene  \\\n",
       "12012  IN-423-TKKC-F-Merged    chr1  1374966  1374966    TG     T     VWA1   \n",
       "12013  IN-423-TKKC-F-Merged    chr1  1581065  1581065     G     C   CDK11B   \n",
       "12014  IN-423-TKKC-F-Merged    chr1  1961365  1961365     G     T    GABRD   \n",
       "12015  IN-423-TKKC-F-Merged    chr1  3102848  3102848     C     T   PRDM16   \n",
       "12016  IN-423-TKKC-F-Merged    chr1  6515380  6515380     G     T     ESPN   \n",
       "\n",
       "      Func.ensGene   ExonicFunc.ensGene  \\\n",
       "12012       exonic  frameshift deletion   \n",
       "12013     intronic                    .   \n",
       "12014     intronic                    .   \n",
       "12015       exonic    nonsynonymous SNV   \n",
       "12016     intronic                    .   \n",
       "\n",
       "                                        AAChange.ensGene Interpro_domain  \\\n",
       "12012  ENSG00000179403:ENST00000404702:exon3:c.501del...               .   \n",
       "12013                                                  .               .   \n",
       "12014                                                  .               .   \n",
       "12015  ENSG00000142611:ENST00000270722:exon2:c.C197T:...               .   \n",
       "12016                                                  .               .   \n",
       "\n",
       "          avsnp150                             CLNDN  \\\n",
       "12012            .                                 .   \n",
       "12013   rs77085542                                 .   \n",
       "12014            .                                 .   \n",
       "12015  rs374012976  Left_ventricular_noncompaction_8   \n",
       "12016   rs75827056                                 .   \n",
       "\n",
       "                          CLNDISDB                 clinvar: Clinvar   Hotspot  \\\n",
       "12012                            .                     clinvar: UNK         0   \n",
       "12013                            .                     clinvar: UNK         0   \n",
       "12014                            .                     clinvar: UNK         0   \n",
       "12015  MedGen:C3809288,OMIM:615373  clinvar: Uncertain_significance         0   \n",
       "12016                            .                     clinvar: UNK         0   \n",
       "\n",
       "      OncoG_TSG  Kinase Cancer Cancer_Short  \n",
       "12012                 0   Lung           LC  \n",
       "12013                 0   Lung           LC  \n",
       "12014                 0   Lung           LC  \n",
       "12015                 0   Lung           LC  \n",
       "12016                 0   Lung           LC  "
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "---"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Pathogenic Variants Subset Creation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Define clinvar filters\n",
    "clinvar_no_conflict = ~df['clinvar: Clinvar '].str.contains(\"conflict\", regex=False, na=False, case=False)\n",
    "clinvar_patho = df['clinvar: Clinvar '].str.contains(\"patho\", regex=False, na=False, case=False)\n",
    "\n",
    "# Define Func.ensGene filter\n",
    "desired_func_values = ['exonic', 'splicing', 'exonic;splicing']\n",
    "func_ensgene_filter = df['Func.ensGene'].isin(desired_func_values)\n",
    "\n",
    "# Combine all conditions\n",
    "patho_only_df = df[clinvar_no_conflict & clinvar_patho & func_ensgene_filter]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "print(patho_only_df['Sample_Name'].nunique())\n",
    "print(patho_only_df.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "patho_only_df.to_clipboard()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "---"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Oncogene"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#--- Pathogenic Oncogene Viz ---#\n",
    "# Subset for oncogenes\n",
    "oncogene_df = patho_only_df[patho_only_df[\"OncoG_TSG\"].isin(['Oncogene', 'Both'])]\n",
    "oncogene_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "print(oncogene_df.shape)\n",
    "oncogene_df['Sample_Name'].nunique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# --- Bar Plot 1: Top Genes ---\n",
    "\n",
    "# Calculate unique sample counts for each gene\n",
    "unique_gene_counts = oncogene_df.groupby('Ref.Gene')['Sample_Name'].nunique()\n",
    "\n",
    "#################################################################################\n",
    "# Calculate the total number of unique samples in the df cohort (1375)\n",
    "total_unique_samples = total_unique_samples_in_df\n",
    "#                                       OR                                      #\n",
    "# Calculate the total number of unique samples in the oncogene_df cohort\n",
    "# total_unique_samples = oncogene_df['Sample_Name'].nunique()\n",
    "#################################################################################\n",
    "\n",
    "# percentage representation based on total unique samples\n",
    "unique_gene_percentage = (unique_gene_counts / total_unique_samples) * 100\n",
    "\n",
    "# Select the top genes based on unique sample counts\n",
    "top_genes_unique = unique_gene_percentage.nlargest(50)\n",
    "top_counts_unique = unique_gene_counts.loc[top_genes_unique.index]  # Actual unique sample counts\n",
    "\n",
    "# Plot\n",
    "plt.figure(figsize=(12, 6))\n",
    "bars = plt.bar(top_genes_unique.index, top_genes_unique.values)\n",
    "\n",
    "# Annotate each bar with percentage (rotated 90 degrees, outside the bar)\n",
    "# for bar, gene in zip(bars, top_genes_unique.index):\n",
    "#     plt.text(bar.get_x() + bar.get_width() / 2, bar.get_height() + 1,  # Shift slightly above\n",
    "#              f'{bar.get_height():.1f}%', ha='center', va='bottom', \n",
    "#              fontsize=10, rotation=90, fontweight='bold')\n",
    "\n",
    "# Modify x-tick labels: Gene name at 90 degrees\n",
    "plt.xticks(range(len(top_genes_unique.index)), top_genes_unique.index, rotation=90, ha='center')\n",
    "\n",
    "plt.ylabel('Sample %')\n",
    "#plt.title('Top 50 Oncogenes by Unique Sample Frequency')\n",
    "\n",
    "# Add grid in the background\n",
    "plt.grid(axis='y', linestyle='--', linewidth=0.7, alpha=0.7)\n",
    "\n",
    "plt.ylim(0, max(top_genes_unique.values) + 2)  # Adjust y-limits to accommodate labels\n",
    "\n",
    "# Show plot\n",
    "plt.show()\n",
    "\n",
    "# Create a corrected table\n",
    "table_oncogene_df = pd.DataFrame({\n",
    "    'Gene': top_genes_unique.index,\n",
    "    'Unique Sample Count': top_counts_unique.values,\n",
    "    'Percentage (%)': top_genes_unique.values\n",
    "})\n",
    "\n",
    "print(table_oncogene_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "table_kinase_df.to_clipboard()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "---"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## TSG"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#--- Pathogenic TSG Viz ---#\n",
    "# Subset for tsg\n",
    "tsg_df = patho_only_df[patho_only_df[\"OncoG_TSG\"].isin(['TSG', 'Both'])]\n",
    "tsg_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "print(tsg_df.shape)\n",
    "tsg_df['Sample_Name'].nunique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# --- Bar Plot 1: Top Genes ---\n",
    "\n",
    "# Calculate unique sample counts for each gene\n",
    "unique_gene_counts = tsg_df.groupby('Ref.Gene')['Sample_Name'].nunique()\n",
    "\n",
    "#################################################################################\n",
    "# Calculate the total number of unique samples in the df cohort (1375)\n",
    "total_unique_samples = total_unique_samples_in_df\n",
    "#                                       OR                                      #\n",
    "# Calculate the total number of unique samples in the tsg_df cohort\n",
    "# total_unique_samples = tsg_df['Sample_Name'].nunique()\n",
    "#################################################################################\n",
    "\n",
    "# percentage representation based on total unique samples\n",
    "unique_gene_percentage = (unique_gene_counts / total_unique_samples) * 100\n",
    "\n",
    "# Select the top genes based on unique sample counts\n",
    "top_genes_unique = unique_gene_percentage.nlargest(50)\n",
    "top_counts_unique = unique_gene_counts.loc[top_genes_unique.index]  # Actual unique sample counts\n",
    "\n",
    "# Plot\n",
    "plt.figure(figsize=(12, 6))\n",
    "bars = plt.bar(top_genes_unique.index, top_genes_unique.values)\n",
    "\n",
    "# Annotate each bar with percentage (rotated 90 degrees, outside the bar)\n",
    "# for bar, gene in zip(bars, top_genes_unique.index):\n",
    "#     plt.text(bar.get_x() + bar.get_width() / 2, bar.get_height() + 1,  # Shift slightly above\n",
    "#              f'{bar.get_height():.1f}%', ha='center', va='bottom', \n",
    "#              fontsize=10, rotation=90, fontweight='bold')\n",
    "\n",
    "# Modify x-tick labels: Gene name at 90 degrees\n",
    "plt.xticks(range(len(top_genes_unique.index)), top_genes_unique.index, rotation=90, ha='center')\n",
    "\n",
    "plt.ylabel('Sample %')\n",
    "#plt.title('Top 50 TSG by Unique Sample Frequency')\n",
    "\n",
    "# Add grid in the background\n",
    "plt.grid(axis='y', linestyle='--', linewidth=0.7, alpha=0.7)\n",
    "\n",
    "plt.ylim(0, max(top_genes_unique.values) + 5)  # Adjust y-limits to accommodate labels\n",
    "\n",
    "# Show plot\n",
    "plt.show()\n",
    "\n",
    "# Create a corrected table\n",
    "table_tsg_df = pd.DataFrame({\n",
    "    'Gene': top_genes_unique.index,\n",
    "    'Unique Sample Count': top_counts_unique.values,\n",
    "    'Percentage (%)': top_genes_unique.values\n",
    "})\n",
    "\n",
    "print(table_tsg_df)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "---"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Kinase"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#--- Pathogenic Kinase Viz ---#\n",
    "# Subset for kinase\n",
    "kinase_df = patho_only_df[patho_only_df[\"Kinase\"].isin([1])]\n",
    "kinase_df.head()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# --- Bar Plot 1: Top Genes ---\n",
    "\n",
    "# Calculate unique sample counts for each gene\n",
    "unique_gene_counts = kinase_df.groupby('Ref.Gene')['Sample_Name'].nunique()\n",
    "\n",
    "#################################################################################\n",
    "# Calculate the total number of unique samples in the df cohort (1375)\n",
    "total_unique_samples = total_unique_samples_in_df\n",
    "#                                       OR                                      #\n",
    "# Calculate the total number of unique samples in the kinase_df cohort\n",
    "# total_unique_samples = kinase_df['Sample_Name'].nunique()\n",
    "#################################################################################\n",
    "\n",
    "# percentage representation based on total unique samples\n",
    "unique_gene_percentage = (unique_gene_counts / total_unique_samples) * 100\n",
    "\n",
    "# Select the top genes based on unique sample counts\n",
    "top_genes_unique = unique_gene_percentage.nlargest(50)\n",
    "top_counts_unique = unique_gene_counts.loc[top_genes_unique.index]  # Actual unique sample counts\n",
    "\n",
    "# Plot\n",
    "plt.figure(figsize=(12, 6))\n",
    "bars = plt.bar(top_genes_unique.index, top_genes_unique.values)\n",
    "\n",
    "# Annotate each bar with percentage (rotated 90 degrees, outside the bar)\n",
    "# for bar, gene in zip(bars, top_genes_unique.index):\n",
    "#     plt.text(bar.get_x() + bar.get_width() / 2, bar.get_height() + 0.5,  # Shift slightly above\n",
    "#              f'{bar.get_height():.1f}%', ha='center', va='bottom', \n",
    "#              fontsize=10, rotation=90, fontweight='bold')\n",
    "\n",
    "# Modify x-tick labels: Gene name at 90 degrees\n",
    "plt.xticks(range(len(top_genes_unique.index)), top_genes_unique.index, rotation=90, ha='center')\n",
    "\n",
    "plt.ylabel('Sample %')\n",
    "#plt.title('Top 50 kinase by Unique Sample Frequency')\n",
    "\n",
    "# Add grid in the background\n",
    "plt.grid(axis='y', linestyle='--', linewidth=0.7, alpha=0.7)\n",
    "\n",
    "plt.ylim(0, max(top_genes_unique.values)+ 1)  # Adjust y-limits to accommodate labels\n",
    "\n",
    "# Show plot\n",
    "plt.show()\n",
    "\n",
    "# Create a corrected table\n",
    "table_kinase_df = pd.DataFrame({\n",
    "    'Gene': top_genes_unique.index,\n",
    "    'Unique Sample Count': top_counts_unique.values,\n",
    "    'Percentage (%)': top_genes_unique.values\n",
    "})\n",
    "\n",
    "print(table_kinase_df)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "---"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Amino Acid mapping"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# # Select columns: 'Ref.Gene', 'AAChange.ensGene', 'avsnp150'\n",
    "# patho_aa_df = patho_only_df.loc[:, ['Ref.Gene', 'avsnp150', 'AAChange.ensGene']]\n",
    "# # Drop duplicate rows across all columns\n",
    "# patho_aa_df = patho_aa_df.drop_duplicates()\n",
    "# # Filter rows based on column: 'AAChange.ensGene' not begining with .\n",
    "# patho_aa_df = patho_aa_df[~patho_aa_df['AAChange.ensGene'].str.startswith(\".\", na=False)]\n",
    "# len(patho_aa_df)\n",
    "# # Extract reference and alternate amino acids\n",
    "# patho_aa_df[\"Ref_AA\"] = patho_aa_df[\"AAChange.ensGene\"].str.extract(r\"p\\.([A-Z])[0-9]+[A-Z]\")[0]\n",
    "# patho_aa_df[\"Alt_AA\"] = patho_aa_df[\"AAChange.ensGene\"].str.extract(r\"p\\.[A-Z][0-9]+([A-Z])\")[0]\n",
    "# patho_aa_df.head()\n",
    "# # Amino acid single-letter to full name mapping\n",
    "# amino_acid_map = {\n",
    "#     \"A\": \"Alanine\", \"R\": \"Arginine\", \"N\": \"Asparagine\", \"D\": \"Aspartic acid\",\n",
    "#     \"C\": \"Cysteine\", \"Q\": \"Glutamine\", \"E\": \"Glutamic acid\", \"G\": \"Glycine\",\n",
    "#     \"H\": \"Histidine\", \"I\": \"Isoleucine\", \"L\": \"Leucine\", \"K\": \"Lysine\",\n",
    "#     \"M\": \"Methionine\", \"F\": \"Phenylalanine\", \"P\": \"Proline\", \"S\": \"Serine\",\n",
    "#     \"T\": \"Threonine\", \"W\": \"Tryptophan\", \"Y\": \"Tyrosine\", \"V\": \"Valine\"\n",
    "# }\n",
    "\n",
    "# # Map single-letter amino acids to full names\n",
    "# patho_aa_df[\"Ref_AA_Full\"] = patho_aa_df[\"Ref_AA\"].map(amino_acid_map)\n",
    "# patho_aa_df[\"Alt_AA_Full\"] = patho_aa_df[\"Alt_AA\"].map(amino_acid_map)\n",
    "# patho_aa_df.head()\n",
    "# # Amino acid group classification bt side chain\n",
    "# amino_acid_groups = {\n",
    "#     \"Aliphatic Amino Acids with Hydrophobic Side Chain\": [\"Alanine\", \"Valine\", \"Leucine\", \"Isoleucine\", \"Methionine\"],\n",
    "#     \"Aromatic Amino Acids with Hydrophobic Side Chain\": [\"Phenylalanine\", \"Tryptophan\", \"Tyrosine\"],\n",
    "#     \"Amino Acids with Neutral Side Chain\": [\"Serine\", \"Threonine\", \"Cysteine\", \"Asparagine\", \"Glutamine\"],\n",
    "#     \"Amino Acids with Negative Charged Side Chain (Acidic)\": [\"Aspartic acid\", \"Glutamic acid\"],\n",
    "#     \"Amino Acids with Positive Charged Side Chain (Basic)\": [\"Lysine\", \"Arginine\", \"Histidine\"],\n",
    "#     \"Imino Acid\": [\"Proline\"],\n",
    "#     \"Unique Amino Acid\": [\"Glycine\"]\n",
    "# }\n",
    "\n",
    "# # Reverse mapping for group assignment\n",
    "# group_mapping = {}\n",
    "# for group, acids in amino_acid_groups.items():\n",
    "#     for acid in acids:\n",
    "#         group_mapping[acid] = group\n",
    "\n",
    "# # Assign groups based on full names\n",
    "# patho_aa_df[\"Ref_Group\"] = patho_aa_df[\"Ref_AA_Full\"].map(group_mapping)\n",
    "# patho_aa_df[\"Alt_Group\"] = patho_aa_df[\"Alt_AA_Full\"].map(group_mapping)\n",
    "\n",
    "# # Determine substitution type\n",
    "# def substitution_type(ref_group, alt_group):\n",
    "#     if ref_group == alt_group:\n",
    "#         return \"Intra\"\n",
    "#     return \"Inter\"\n",
    "\n",
    "# patho_aa_df[\"Substitution_Type\"] = patho_aa_df.apply(\n",
    "#     lambda x: substitution_type(x[\"Ref_Group\"], x[\"Alt_Group\"]), axis=1\n",
    "# )\n",
    "\n",
    "# # Display the DataFrame\n",
    "# print(patho_aa_df)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "---"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## All Genes Cancer Distribution"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Cancer Counts\n",
    "patho_only_df.groupby('Cancer_Short')['Sample_Name'].nunique().reset_index(name='Unique_Sample_Count')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "patho_only_cancercounts = patho_only_df.groupby('Cancer_Short')['Sample_Name'].nunique().reset_index(name='Unique_Sample_Count')\n",
    "patho_only_cancercounts.to_clipboard()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "patho_only_df.groupby('Cancer_Short')['Ref.Gene'].nunique().reset_index(name='Gene_Count')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "patho_only_df.groupby('Ref.Gene')['Sample_Name'].nunique().reset_index(name='Sample_Count').sort_values(by='Sample_Count', ascending=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Count the number of samples each gene is present in\n",
    "gene_sample_counts = patho_only_df.groupby(\"Ref.Gene\")[\"Sample_Name\"].nunique()\n",
    "\n",
    "# Find the most common cancer type for each gene\n",
    "gene_cancer_counts = patho_only_df.groupby([\"Ref.Gene\", \"Cancer_Short\"])[\"Sample_Name\"].nunique()\n",
    "most_common_cancer_per_gene = gene_cancer_counts.groupby(level=0).idxmax().apply(lambda x: x[1])\n",
    "\n",
    "# Combine results into a dataframe\n",
    "gene_summary = pd.DataFrame({\n",
    "    \"Sample_Count\": gene_sample_counts,\n",
    "    \"Most_Common_Cancer\": most_common_cancer_per_gene\n",
    "})\n",
    "\n",
    "print(gene_summary)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_summary.to_clipboard()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Count the number of samples each gene is present in\n",
    "gene_sample_counts = patho_only_df.groupby(\"Ref.Gene\")[\"Sample_Name\"].nunique()\n",
    "\n",
    "# Find the most common cancer type for each gene\n",
    "gene_cancer_counts = patho_only_df.groupby([\"Ref.Gene\", \"Cancer_Short\"])[\"Sample_Name\"].nunique()\n",
    "most_common_cancer_per_gene = gene_cancer_counts.groupby(level=0).idxmax().apply(lambda x: x[1])\n",
    "\n",
    "# Combine results into a dataframe\n",
    "gene_summary = pd.DataFrame({\n",
    "    \"Sample_Count\": gene_sample_counts,\n",
    "    \"Most_Common_Cancer\": most_common_cancer_per_gene\n",
    "})\n",
    "\n",
    "# Select top genes based on sample count\n",
    "top_genes = gene_summary.nlargest(50, \"Sample_Count\")\n",
    "\n",
    "# Calculate percentage sample count\n",
    "top_genes[\"Sample_Percentage\"] = (top_genes[\"Sample_Count\"] / 1375) * 100\n",
    "\n",
    "# Increase figure size for better readability\n",
    "plt.figure(figsize=(12, 6))\n",
    "bars = plt.bar(top_genes.index, top_genes[\"Sample_Percentage\"], color=\"skyblue\")\n",
    "\n",
    "# Annotate bars with most common cancer type\n",
    "for bar, cancer in zip(bars, top_genes[\"Most_Common_Cancer\"]):\n",
    "    plt.text(bar.get_x() + bar.get_width()/2, bar.get_height() +1, cancer, ha=\"center\", va=\"bottom\", fontsize=11, rotation=90, fontweight='bold')\n",
    "\n",
    "plt.ylim(0, max(top_genes['Sample_Percentage']) + 5)  # Adjust y-limits to accommodate labels\n",
    "\n",
    "# Formatting\n",
    "plt.xlabel(\"Gene\")\n",
    "plt.ylabel(\"Percentage of Samples\")\n",
    "#plt.title(\"Top 50 Genes by Sample Percentage with Most Common Cancer Type (Sample Count = 1375)\")\n",
    "plt.xticks(rotation=90, fontsize=10)  # Rotate x-axis labels for better readability\n",
    "plt.yticks(fontsize=10)\n",
    "plt.grid(axis=\"y\", linestyle=\"--\", alpha=0.7)\n",
    "plt.tight_layout()\n",
    "\n",
    "# Show plot\n",
    "plt.show()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Count the number of samples per cancer type\n",
    "cancer_sample_counts = df.groupby(\"Cancer_Short\")[\"Sample_Name\"].nunique().sort_values(ascending=False)\n",
    "\n",
    "# Set up figure\n",
    "fig, ax = plt.subplots(figsize=(14, 4))\n",
    "\n",
    "# Generate colors for bars\n",
    "colors = plt.cm.tab20.colors  # Using tab20 colormap for variety\n",
    "\n",
    "# Create bar plot\n",
    "bars = ax.bar(\n",
    "    cancer_sample_counts.index, \n",
    "    cancer_sample_counts.values, \n",
    "    color=[colors[i % len(colors)] for i in range(len(cancer_sample_counts))], \n",
    "    edgecolor='black'\n",
    ")\n",
    "\n",
    "# Annotate bars with sample counts\n",
    "for bar, count in zip(bars, cancer_sample_counts.values):\n",
    "    ax.text(\n",
    "        bar.get_x() + bar.get_width()/2, \n",
    "        bar.get_height(), \n",
    "        str(count), \n",
    "        ha=\"center\", \n",
    "        va=\"bottom\", \n",
    "        fontsize=9, \n",
    "        fontweight=\"bold\"\n",
    "    )\n",
    "\n",
    "# Formatting\n",
    "ax.set_xlabel(\"Cancer Type\", fontsize=12, fontweight=\"bold\")\n",
    "ax.set_ylabel(\"# of Samples\", fontsize=12, fontweight=\"bold\")\n",
    "#ax.set_title(\"Sample Distribution per Cancer Type\", fontsize=14, fontweight=\"bold\")\n",
    "ax.set_xticks(np.arange(len(cancer_sample_counts.index)))\n",
    "ax.set_xticklabels(cancer_sample_counts.index, rotation=90, fontsize=9)\n",
    "ax.grid(axis=\"y\", linestyle=\"--\", alpha=0.7)\n",
    "ax.set_axisbelow(True)\n",
    "\n",
    "# Increase y-axis limit for better spacing\n",
    "ax.set_ylim(0, max(cancer_sample_counts.values) * 1.1)  # Add 10% extra space\n",
    "\n",
    "# Adjust layout for better fit\n",
    "plt.tight_layout()\n",
    "\n",
    "# Show plot\n",
    "plt.show()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Count the number of samples per cancer type\n",
    "cancer_sample_counts = patho_only_df.groupby(\"Cancer_Short\")[\"Sample_Name\"].nunique().sort_values(ascending=False)\n",
    "\n",
    "# Set up figure\n",
    "fig, ax = plt.subplots(figsize=(15, 5))\n",
    "\n",
    "# Generate colors for bars\n",
    "colors = plt.cm.tab20.colors  # Using tab20 colormap for variety\n",
    "\n",
    "# Create bar plot\n",
    "bars = ax.bar(\n",
    "    cancer_sample_counts.index, \n",
    "    cancer_sample_counts.values, \n",
    "    color=[colors[i % len(colors)] for i in range(len(cancer_sample_counts))], \n",
    "    edgecolor='black'\n",
    ")\n",
    "\n",
    "# Annotate bars with sample counts\n",
    "for bar, count in zip(bars, cancer_sample_counts.values):\n",
    "    ax.text(\n",
    "        bar.get_x() + bar.get_width()/2, \n",
    "        bar.get_height(), \n",
    "        str(count), \n",
    "        ha=\"center\", \n",
    "        va=\"bottom\", \n",
    "        fontsize=9, \n",
    "        fontweight=\"bold\"\n",
    "    )\n",
    "\n",
    "# Formatting\n",
    "ax.set_xlabel(\"Cancer Type\", fontsize=12, fontweight=\"bold\")\n",
    "ax.set_ylabel(\"# of Samples\", fontsize=12, fontweight=\"bold\")\n",
    "#ax.set_title(\"Sample Distribution per Cancer Type in Pathogenic only Subset\", fontsize=14, fontweight=\"bold\")\n",
    "ax.set_xticks(np.arange(len(cancer_sample_counts.index)))\n",
    "ax.set_xticklabels(cancer_sample_counts.index, rotation=90, fontsize=9)\n",
    "ax.grid(axis=\"y\", linestyle=\"--\", alpha=0.7)\n",
    "ax.set_axisbelow(True)\n",
    "# Increase y-axis limit for better spacing\n",
    "ax.set_ylim(0, max(cancer_sample_counts.values) * 1.1)  # Add 10% extra space\n",
    "\n",
    "\n",
    "# Adjust layout for better fit\n",
    "plt.tight_layout()\n",
    "\n",
    "# Show plot\n",
    "plt.show()\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "---"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# --- Calculate the percentage of samples with mutant genes per cancer type --- #\n",
    "\n",
    "# Count the number of unique samples per cancer type\n",
    "unique_samples_per_cancer = patho_only_df.groupby('Cancer_Short')['Sample_Name'].nunique()\n",
    "# Count the number of mutations per cancer type\n",
    "mutation_counts = patho_only_df['Cancer_Short'].value_counts()\n",
    "# Count the number of unique genes per cancer type\n",
    "mutant_gene_counts = patho_only_df.groupby('Cancer_Short')['Ref.Gene'].nunique()\n",
    "\n",
    "\n",
    "# Calculate the percentage\n",
    "gene_by_mutation = (mutant_gene_counts / mutation_counts) * 100\n",
    "mutation_by_gene = (mutation_counts / mutant_gene_counts) * 100\n",
    "# Calculate the percentage of unique samples with mutant genes per cancer type\n",
    "gene_by_sample = (mutant_gene_counts / unique_samples_per_cancer) * 100\n",
    "\n",
    "mutation_by_sample = (mutation_counts / unique_samples_per_cancer ) * 100\n",
    "\n",
    "# Identify the most frequently mutated gene for each cancer type\n",
    "most_mutated_gene = patho_only_df.groupby('Cancer_Short')['Ref.Gene'].agg(lambda x: x.value_counts().idxmax())\n",
    "\n",
    "# Combine the results into a single DataFrame\n",
    "mutation_summary = pd.DataFrame({\n",
    "    'Total Unique Samples': unique_samples_per_cancer,\n",
    "    'Total Mutations': mutation_counts,\n",
    "    'Unique Mutant Genes': mutant_gene_counts,\n",
    "    'Gene/Mutations': gene_by_mutation,\n",
    "    'Mutations/Gene': mutation_by_gene,\n",
    "    'Genes/Samples': gene_by_sample,\n",
    "    'Muatations/Samples': mutation_by_sample,\n",
    "    'Most Mutated Gene': most_mutated_gene\n",
    "}).reset_index().rename(columns={'index': 'Cancer_Short'})\n",
    "\n",
    "mutation_summary.head(n=50)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# --- Calculate the percentage of samples with mutant genes per cancer type --- #\n",
    "\n",
    "# Count the number of unique samples per cancer type\n",
    "unique_samples_per_cancer_df = df.groupby('Cancer_Short')['Sample_Name'].nunique()\n",
    "# Count the number of unique samples per cancer type\n",
    "unique_samples_per_cancer_patho_df = patho_only_df.groupby('Cancer_Short')['Sample_Name'].nunique()\n",
    "\n",
    "# Identify the most frequently mutated gene for each cancer type\n",
    "most_mutated_patho_gene = patho_only_df.groupby('Cancer_Short')['Ref.Gene'].agg(lambda x: x.value_counts().idxmax())\n",
    "\n",
    "# Combine the results into a single DataFrame\n",
    "mutation_summary = pd.DataFrame({\n",
    "    'Total Unique Samples': unique_samples_per_cancer_df,\n",
    "    'Total Unique Samples having Pathogenic Mutation': unique_samples_per_cancer_patho_df,\n",
    "    'Most Mutated Gene': most_mutated_patho_gene\n",
    "}).reset_index().rename(columns={'index': 'Cancer_Short'})\n",
    "\n",
    "mutation_summary.head(n=50)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "mutation_summary.to_clipboard()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Update the bar chart to annotate with most mutated gene\n",
    "df_updated = mutation_summary\n",
    "\n",
    "# Calculate the percentage of samples mutated\n",
    "df_updated[\"Percentage Mutated\"] = (df_updated[\"Total Unique Samples having Pathogenic Mutation\"] / df_updated[\"Total Unique Samples\"]) * 100\n",
    "\n",
    "# Plot the bar chart\n",
    "plt.figure(figsize=(12, 6))\n",
    "bars = plt.barh(df_updated[\"Cancer_Short\"], df_updated[\"Percentage Mutated\"], color=\"steelblue\")\n",
    "plt.xlabel(\"Percentage of Samples Mutated\")\n",
    "plt.ylabel(\"Cancer Type\")\n",
    "#plt.title(\"Percentage of Unique Samples Having Pathogenic Mutation per Cancer Type with Most Mutated Gene labelled\")\n",
    "\n",
    "# Annotate bars with the most mutated gene\n",
    "for bar, gene in zip(bars, df_updated[\"Most Mutated Gene\"]):\n",
    "    plt.text(bar.get_width() + 1, bar.get_y() + bar.get_height()/2, gene, va='center')\n",
    "\n",
    "plt.gca().invert_yaxis()  # Invert y-axis for better readability\n",
    "\n",
    "# Add vertical grid lines\n",
    "plt.grid(axis='x', linestyle='--', alpha=0.4)\n",
    "\n",
    "# Display the plot\n",
    "plt.show()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# # Step 1: Count the number of samples per cancer type\n",
    "# total_samples_per_cancer = patho_only_df.groupby(\"Cancer\")[\"Sample_Name\"].nunique().reset_index()\n",
    "# total_samples_per_cancer.columns = [\"Cancer\", \"Total_Samples\"]\n",
    "\n",
    "# # Step 2: Count mutations per gene in each cancer type\n",
    "# mutation_counts = patho_only_df.groupby([\"Cancer\", \"Ref.Gene\"])[\"Sample_Name\"].nunique().reset_index()\n",
    "# mutation_counts.columns = [\"Cancer\", \"Gene\", \"Mutated_Samples\"]\n",
    "\n",
    "# # Step 3: Merge to get total sample count per cancer type\n",
    "# mutation_freq = mutation_counts.merge(total_samples_per_cancer, on=\"Cancer\")\n",
    "\n",
    "# # Step 4: Calculate mutation frequency\n",
    "# mutation_freq[\"Mutation_Frequency\"] = (mutation_freq[\"Mutated_Samples\"] / mutation_freq[\"Total_Samples\"]) * 100\n",
    "\n",
    "# # Step 5: Plot the mutation frequencies\n",
    "# plt.figure(figsize=(20, 6))\n",
    "# sns.stripplot(data=mutation_freq, x=\"Cancer\", y=\"Mutation_Frequency\", hue=\"Gene\", jitter=True, alpha=0.7)\n",
    "\n",
    "# # Formatting\n",
    "# plt.xticks(rotation=90)\n",
    "# plt.xlabel(\"Cancer Type\")\n",
    "# plt.ylabel(\"% of Samples with a Mutant Gene\")\n",
    "# plt.title(\"Mutations in Cancer Type\")\n",
    "# plt.legend([], [], frameon=False)  # Hide legend for clarity\n",
    "# # plt.legend(title=\"Gene\", bbox_to_anchor=(1.05, 1), loc='upper left')\n",
    "\n",
    "# # Show the plot\n",
    "# plt.show()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Count the number of unique samples per cancer type\n",
    "unique_samples_per_cancer = patho_only_df.groupby('Cancer')['Sample_Name'].nunique()\n",
    "\n",
    "# Count the number of unique mutant genes per cancer type\n",
    "mutant_gene_counts = patho_only_df.groupby('Cancer')['Ref.Gene'].nunique()\n",
    "\n",
    "# Identify the most frequently mutated gene for each cancer type\n",
    "most_mutated_gene = patho_only_df.groupby('Cancer')['Ref.Gene'].agg(lambda x: x.value_counts().idxmax())\n",
    "\n",
    "# Create a summary DataFrame\n",
    "mutation_summary_corrected = pd.DataFrame({\n",
    "    'Total Unique Samples': unique_samples_per_cancer,\n",
    "    'Unique Mutant Genes': mutant_gene_counts,\n",
    "    'Most Mutated Gene': most_mutated_gene\n",
    "}).reset_index().rename(columns={'index': 'Cancer'})\n",
    "\n",
    "#Plot\n",
    "plt.figure(figsize=(12, 6))\n",
    "bars = plt.bar(mutation_summary_corrected['Cancer'], mutation_summary_corrected['Unique Mutant Genes'])\n",
    "\n",
    "# Add labels for the most mutated gene on top of each bar\n",
    "for bar, gene in zip(bars, mutation_summary_corrected['Most Mutated Gene']):\n",
    "    plt.text(bar.get_x() + bar.get_width() / 2, bar.get_height(), gene, ha='center', va='bottom', fontsize=10, fontweight='bold')\n",
    "\n",
    "# Customize the plot\n",
    "plt.xlabel('Cancer Type')\n",
    "plt.ylabel('Number of Mutations')\n",
    "plt.title('Number of Mutations per Cancer Type with Top Mutated Gene')\n",
    "plt.xticks(rotation=45, ha='right')\n",
    "plt.grid(axis='y', linestyle='--', alpha=0.7)\n",
    "\n",
    "# Show the plot\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "---"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Count unique samples per gene and Cancer type\n",
    "gene_mutation_counts = patho_only_df.groupby([\"Ref.Gene\", \"Cancer_Short\"])[\"Sample_Name\"].nunique().reset_index()\n",
    "gene_mutation_counts.columns = [\"Gene\", \"Cancer_Short\", \"Unique Mutation Count\"]\n",
    "\n",
    "# Pivot the data for heatmap plotting\n",
    "heatmap_data = gene_mutation_counts.pivot(index=\"Gene\", columns=\"Cancer_Short\", values=\"Unique Mutation Count\")\n",
    "\n",
    "# Select the top 30 genes with the highest mutation counts\n",
    "top_genes = heatmap_data.sum(axis=1).sort_values(ascending=False).head(30).index\n",
    "heatmap_data_filtered = heatmap_data.loc[top_genes]\n",
    "\n",
    "# Plot the improved heatmap\n",
    "plt.figure(figsize=(15, 8))\n",
    "sns.heatmap(heatmap_data_filtered, cmap=\"Blues\", linewidths=0.5, annot=True, fmt=\".0f\")\n",
    "\n",
    "# Set labels and title\n",
    "plt.xlabel(\"Cancer Type\", fontsize=12)\n",
    "plt.ylabel(\"Pathogenic Genes\", fontsize=12)\n",
    "plt.title(\"Top 30 Mutated Genes Across Cancer Types\", fontsize=14)\n",
    "\n",
    "# Rotate x-axis labels for readability\n",
    "plt.xticks(rotation=90)\n",
    "plt.yticks(rotation=0)\n",
    "\n",
    "# Display the refined heatmap\n",
    "plt.show()\n",
    "\n",
    "# Display the heatmap\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Count unique samples per gene and cancer type\n",
    "gene_mutation_counts = patho_only_df.groupby([\"Ref.Gene\", \"Cancer_Short\"])[\"Sample_Name\"].nunique().reset_index()\n",
    "gene_mutation_counts.columns = [\"Gene\", \"Cancer_Short\", \"Unique Mutation Count\"]\n",
    "\n",
    "# Count total unique samples per cancer type\n",
    "total_samples_per_cancer_df = patho_only_df.groupby(\"Cancer_Short\")[\"Sample_Name\"].nunique().reset_index()\n",
    "total_samples_per_cancer_df.columns = [\"Cancer_Short\", \"Total Unique Samples\"]\n",
    "\n",
    "# Merge the mutation counts with total sample counts\n",
    "gene_mutation_percentage_df = gene_mutation_counts.merge(total_samples_per_cancer_df, on=\"Cancer_Short\")\n",
    "\n",
    "# Calculate the mutation percentage per gene and cancer type\n",
    "gene_mutation_percentage_df[\"Mutation Percentage\"] = (\n",
    "    gene_mutation_percentage_df[\"Unique Mutation Count\"] / gene_mutation_percentage_df[\"Total Unique Samples\"]\n",
    ") * 100\n",
    "\n",
    "# Pivot the data for heatmap plotting\n",
    "heatmap_data = gene_mutation_percentage_df.pivot(index=\"Gene\", columns=\"Cancer_Short\", values=\"Mutation Percentage\")\n",
    "\n",
    "# Select the top 30 genes with the highest mutation counts\n",
    "top_genes = heatmap_data.sum(axis=1).sort_values(ascending=False).head(30).index\n",
    "heatmap_data_filtered = heatmap_data.loc[top_genes]\n",
    "\n",
    "# Plot the improved heatmap\n",
    "plt.figure(figsize=(15, 8))\n",
    "sns.heatmap(heatmap_data_filtered, cmap=\"Blues\", linewidths=0.5, annot=True, fmt=\".0f\")\n",
    "\n",
    "# Set labels and title\n",
    "plt.xlabel(\"Cancer Type\", fontsize=12)\n",
    "plt.ylabel(\"Pathogenic Genes\", fontsize=12)\n",
    "#plt.title(\"Top 30 Pathogenic Genes Across Cancer Types with counts as percentage\", fontsize=14)\n",
    "\n",
    "# Rotate x-axis labels for readability\n",
    "plt.xticks(rotation=90)\n",
    "plt.yticks(rotation=0)\n",
    "\n",
    "# Display the refined heatmap\n",
    "plt.show()\n",
    "\n",
    "# Display the heatmap\n",
    "plt.show()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "heatmap_data_filtered"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "---"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# --- For plotting the ExonicFunc.ensGene distribution per Cancer type --- #\n",
    "\n",
    "# Filter out rows where ExonicFunc.ensGene is \".\"\n",
    "filtered_df = patho_only_df[df[\"ExonicFunc.ensGene\"] != \".\"]\n",
    "\n",
    "# Count occurrences of each ExonicFunc.ensGene per Cancer type\n",
    "filtered_exonic_func_counts = filtered_df.groupby([\"Cancer_Short\", \"ExonicFunc.ensGene\"]).size().unstack(fill_value=0)\n",
    "\n",
    "# Increase figure size for better label alignment\n",
    "fig, ax = plt.subplots(figsize=(12, 6))\n",
    "\n",
    "# Define color mapping using Set2 qualitative colormap\n",
    "set2_colors = cm.get_cmap(\"Set2\", len(filtered_exonic_func_counts.columns)).colors\n",
    "color_map = {func: set2_colors[i % len(set2_colors)] for i, func in enumerate(filtered_exonic_func_counts.columns)}\n",
    "\n",
    "# Plot the stacked bar chart with Set2 colormap\n",
    "filtered_exonic_func_counts.plot(kind=\"bar\", stacked=True, color=[color_map[func] for func in filtered_exonic_func_counts.columns], ax=ax)\n",
    "\n",
    "# Improve layout and readability\n",
    "plt.xlabel(\"Cancer Type\")\n",
    "plt.ylabel(\"Number of Variants\")\n",
    "#plt.title(\"Distribution of ExonicFunc.ensGene per Cancer Type (Excluding '.')\")\n",
    "plt.xticks(rotation=45, ha='right')  # Align x-axis labels properly\n",
    "plt.legend(title=\"Variant Consequences\", bbox_to_anchor=(1, 1))\n",
    "plt.tight_layout()\n",
    "\n",
    "# Add grid lines\n",
    "plt.grid(axis='y', linestyle='--', alpha=0.4)\n",
    "\n",
    "# Set y-axis ticks at intervals\n",
    "ax.set_yticks(np.arange(0, filtered_exonic_func_counts.sum(axis=1).max() + 50, 50))\n",
    "\n",
    "# Show the plot\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "filtered_exonic_func_counts.head(n=20)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Calculate the total for each Cancer type\n",
    "filtered_exonic_func_counts[\"Total\"] = filtered_exonic_func_counts.sum(axis=1)\n",
    "\n",
    "# Compute the percentage for each exonic term\n",
    "percentage_filtered_exonic_func_counts = filtered_exonic_func_counts.div(filtered_exonic_func_counts[\"Total\"], axis=0) * 100\n",
    "\n",
    "# Drop the \"Total\" column after computing percentages\n",
    "percentage_filtered_exonic_func_counts.drop(columns=[\"Total\"], inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Increase figure size for better label alignment\n",
    "fig, ax = plt.subplots(figsize=(12, 6))\n",
    "\n",
    "# Define color mapping using Set2 qualitative colormap\n",
    "set2_colors = cm.get_cmap(\"Set2\", len(percentage_filtered_exonic_func_counts.columns)).colors\n",
    "color_map = {func: set2_colors[i % len(set2_colors)] for i, func in enumerate(percentage_filtered_exonic_func_counts.columns)}\n",
    "\n",
    "# Plot the stacked bar chart with Set2 colormap\n",
    "percentage_filtered_exonic_func_counts.plot(kind=\"bar\", stacked=True, color=[color_map[func] for func in percentage_filtered_exonic_func_counts.columns], ax=ax)\n",
    "\n",
    "# Improve layout and readability\n",
    "plt.xlabel(\"Cancer Type\")\n",
    "plt.ylabel(\"Number of Variants %\")\n",
    "#plt.title(\"Unique ExonicFunc.ensGene Counts by Pathogenic Gene per Cancer Type\")\n",
    "plt.xticks(rotation=45, ha='right')  # Align x-axis labels properly\n",
    "plt.legend(title=\"Variant Consequences\", bbox_to_anchor=(1, 1))\n",
    "plt.tight_layout()\n",
    "\n",
    "# Set y-axis ticks at intervals of 5\n",
    "ax.set_yticks(np.arange(0, percentage_filtered_exonic_func_counts.sum(axis=1).max(), 5))\n",
    "\n",
    "# Add grid lines\n",
    "plt.grid(axis='y', linestyle='--', alpha=0.4)\n",
    "\n",
    "# Show the plot\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Select columns: 'Sample_Name', 'Cancer'\n",
    "patho_samp = patho_only_df.loc[:, ['Sample_Name', 'Cancer']]\n",
    "patho_samp = patho_samp.drop_duplicates()\n",
    "patho_samp = patho_samp.rename(columns={'Sample_Name': 'Sample'})\n",
    "patho_samp = patho_samp.rename(columns={'Cancer': 'Cohort'})\n",
    "print(patho_samp)\n",
    "patho_samp.to_csv(r\"H:\\My Drive\\Pathogenic_Landscape\\data\\absolute\\clinical_research_filtered_combined\\dataframes\\patho_samp.tsv\", sep=\"\\t\", index= False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "print(patho_only_df.shape)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "base",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
